src/java/org/apache/cassandra/security/DefaultCryptoProvider.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.security;
+
+import java.util.Map;
+import javax.crypto.Cipher;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.cassandra.exceptions.StartupException;
+import com.amazon.corretto.crypto.provider.AmazonCorrettoCryptoProvider;
+
+public class DefaultCryptoProvider implements ICryptoProvider
+{
+    private static final Logger logger = LoggerFactory.getLogger(DefaultCryptoProvider.class);
+
+    public DefaultCryptoProvider(Map<String, String> args) {}
+    @Override
+    public void installProvider() throws StartupException
```

Comment:
This method does not throw any `StartupException`, does it? If you are installing the provider as part of `applyConfig`, if you catch `Exception`, you should wrap it in `ConfigurationException` and throw that one instead.

Reply:
It was done intentionally for cases where ACCP was not installed properly. We don't want to stop Cassandra from booting up, so it will use another provider as a fallback: "sun.security.provider.Sun". I have added exception to the interface to give flexibility to others in how they want to handle the implementations of their custom provider. I will update it to throw Exception and not StartupException.

Reply:
Interesting. I think that if we fail the installation of the provider, we should not just silently continue, we should truly fail the startup. A user asked for a crypto_provider he did not get.

Reply:
100% yes, this is a pet peeve of mine. Something is not configured right and the startup continues and you will end up with a Cassandra which is substantially slower. If necessary make it an option so people can configure it the way they need it.